### ISSUE TITLE
Automatic link for macro defined in different header file no longer generated

### ISSUE THREAD

**Issue by davidritter** [doxygen/doxygen#10994]
**Describe the bug**
In Doxygen 1.8.9.1 the link for the macro would be genereated. With Doxygen 1.11.0 it is not. 

**Screenshots**
Doxygen 1.8.9.1
![image](https://github.com/doxygen/doxygen/assets/24844710/81f57e91-7245-44d6-ade3-50eddc8a5564)

Doxygen 1.11.0
![image](https://github.com/doxygen/doxygen/assets/24844710/0dbb7d0f-bc80-4e59-b1e3-5499d488fccf)

**To Reproduce**
[testcase.zip](https://github.com/user-attachments/files/16177922/testcase.zip)

**Expected behavior**
Automatic linking should occur.

**Version**
`doxygen 1.11.0` on Linux

**Additional context**
I can work around this by using `\ref` like:
```
\ref B::BCALLBACK(CALLBACK) "BCALLBACK(CALLBACK)"
```

**Comment by albert-github** [comment]
The difference has been introduced in doxygen 1.9.5 and more precise by means of:
```
Commit: b290399fab92df45376103d7c094f053f24495fa [b290399]

Date: Tuesday, May 24, 2022 8:12:07 AM

Commit Date: Friday, June 3, 2022 9:38:54 PM
Refactoring: reimplement getDefs using symbol resolver
```

**Comment by albert-github** [comment]
Might be related to https://github.com/doxygen/doxygen/issues/10915 as this points to the same commit resulting in the difference.

**Comment by davidritter** [comment]
Using the MSVC release build artifact from this CI job 
https://github.com/doxygen/doxygen/actions/runs/9944580078 for the commit I do see the link being automatically generated again.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.12.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
